### PR TITLE
Update project since mypy is introduced

### PR DESCRIPTION
**Pull request by jxltom**
Since mypy in introduced in https://github.com/sarugaku/pythonfinder/pull/51, this PR update requirements, add ```mypy``` to pipenv's scripts and update relevant ```.gitignore```

It is blocked by https://github.com/sarugaku/pythonfinder/pull/53 and #51 for now

### CODE REVIEW

**Comment by jxltom** [comment]
I'm curious why ```typed-ast``` needs markers manually? Why pipenv can not add markers automatically... 😕

**Comment by techalchemy** [comment]
Mypy itself doesn’t require markers does it? You just need `typing; python_version<3.5`

**Comment by jxltom** [comment]
Actually mypy required python 3.4+, otherwise pipenv will fail on locking

Check here https://github.com/python/mypy/blob/master/setup.py#L8-L10

**Comment by techalchemy** [comment]
Ah yeah. Anyway we considered cascading markers down. Now that we have a way to check reverse dependencies we can do that

**Comment by jxltom** [comment]
That will be great

File: Pipfile.lock [hunk 1]
```diff
@@ -72,30 +64,6 @@
             ],
             "version": "==2.6.0"
         },
-        "backports.functools-lru-cache": {
```
Comment (techalchemy) [inline Pipfile.lock]: Now do you see why i want `--keep-outdated` to respect things that used to be in the pipfile even if we can't find a parent for them anymore?  This is a valid dependency!

Comment (jxltom) [reply Pipfile.lock]: Indeed. But I'm curious why it is removed. I'm using python3 within the virtualenv and locking. Maybe you are using python2?

Comment (techalchemy) [reply Pipfile.lock]: I generate and update with both versions typically to make sure I don’t have failing builds, which is why I focused on the use case of not removing dependencies that the resolver thinks are invalid

Comment (jxltom) [reply Pipfile.lock]: Understood, feels the same now

**Pull request merged**